crates/bevy_window/src/window.rs

```diff
@@ -108,6 +108,27 @@ pub enum WindowCommand {
     },
 }
 
+#[derive(Debug, Clone)]
+pub struct Icon {
+    pub rgba: Vec<u8>,
+    pub width: u32,
+    pub height: u32,
+}
+
+impl Icon {
+    /// Creates an `Icon` from 32bpp RGBA data.
+    ///
+    /// The length of `rgba` must be divisible by 4, and `width * height` must equal
+    /// `rgba.len() / 4`. Otherwise, this will icon will not be used.
+    pub fn from_rgba(rgba: Vec<u8>, width: u32, height: u32) -> Self {
```

Comment:
Ideally we'd support create an icon from any image type supported by the `image` crate. I'm not sure which crate that support should live in, however.